tools/ai_sidecar/server.py

```
@@ -276,8 +414,22 @@ def main() -> None:
         _run_compare_laps(args.compare_laps[0], args.compare_laps[1])
         return
     reply = not args.no_reply
+
+    if args.external_bind is not None:
+        if not args.token:
+            raise SystemExit("--external-bind requires --token (refusing to expose unauthenticated socket)")
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)